server/schedule/checker/rule_checker.go

```diff
@@ -166,6 +172,9 @@ func (c *RuleChecker) addRulePeer(region *core.RegionInfo, rf *placement.RuleFit
 		checkerCounter.WithLabelValues("rule_checker", "no-store-add").Inc()
 		if filterByTempState {
 			c.regionWaitingList.Put(region.GetID(), nil)
+			c.pendingList.Remove(region.GetID())
+		} else {
+			c.pendingList.Put(region.GetID(), nil)
 		}
```

Comment:
These operations can be extracted as a function since they always work together.